### DISCUSSION TITLE
Request for Guidance: Configuring cgroup v1 and AppArmor in NEW U-Boot

### DISCUSSION THREAD

**Discussion by santrixever** [devmfc/debian-on-amlogic#84] [Q&A] [answered]
Hi devmfc and the community,

First of all, I want to thank you for the amazing repository you’ve built and for your contributions to the open-source community. It’s truly impressive!

I have an X96mini with an Amlogic S905W2 with 2GB of RAM, and to run the operating system, I had to use the box=t95w2_s905y4 option. Although I’m not an expert in Linux, I’ve been trying to solve an issue on my own before reaching out.

What I need is to apply the following configuration in U-Boot: apparmor=1 security=apparmor systemd.unified_cgroup_hierarchy=0, especially to use cgroup v1. In other systems, I’ve added these parameters directly in the APPEND section within U-Bootbut I haven’t been able to do that here since I couldn’t find that option.

I’ve added the following in the bootscript:

```# Configuration for cgroup v1
setenv cgroup_v1 "cgroup_enable=cpuset cgroup_enable=memory"
setenv bootargs "${bootargs} ${cgroup_v1}"
```

And I’ve regenerated the compiled files for both bootscript and boot.scr, but I haven’t been successful.

I really appreciate your time and any help you can provide.

Thank you in advance!

**Comment by devmfc** [comment, marked as answer]
You're welcome.

Just add `bootargs0=cgroup_enable=cpuset cgroup_enable=memory` to boot.config. You can also use bootargs1, 2, 3 etc.
But what you did should also work?
